test/test.js

```diff
@@ -262,19 +262,18 @@ describe('hotkey', function () {
 
   describe('eventToHotkeyString', function () {
     const tests = [
-      ['Control+J', {ctrlKey: true, shiftKey: true, code: 'KeyJ', key: 'J'}],
-      ['Control+Shift+j', {ctrlKey: true, shiftKey: true, code: 'KeyJ', key: 'j'}],
```

Comment:
This test doesn't really make sense - it's `Shift` but with a lowercase letter. I deleted it.

Reply:
it looks like maybe this can be traced to https://github.com/github/hotkey/commit/a8e22eeddeb911dd007eaffdcb3a6372d8628cf7 ?

Although that test should be `Meta+Shift+j`? I see there's a test for that below on line 268. So maybe this is just there for completeness.

Reply:
At any rate this does bring up an important point: are we sure that `Mod+Shift+M` will work consistantly across platforms?

We may want to lowercase the keynames or something?

Reply:
> This test doesn't really make sense - it's Shift but with a lowercase letter. I deleted it.

This probably could have had a comment to describe why it's necessary - but it is in the commit information Ned points out. The (incorrect) assumption here is that a Shift key turns printable characters into their uppercase variants, however the Shift key does not always mean that the casing of the printable key will be uppercase. See for e.g. https://bugs.webkit.org/show_bug.cgi?id=174782. This issue is complex because if we try to coerce macos to do the right thing, it means some character shortcuts cannot be represented.

Reply:
Is that true for Linux? https://github.github.com/hotkey/hotkey_mapper.html is broken for me it seems but I can visit https://www.toptal.com/developers/keycode on a linux machine and press `Ctrl+Shift+M` and it comes out uppercase, and the `ctrlKey` is `true` which would map to `Mod` in this codebase, if I'm reading correctly.

Reply:
Oh man, I hope so 😬 

Would you mind doing that [here](https://w3c.github.io/uievents/tools/key-event-viewer.html) and posting a screenshot? I don't have a linux machine to try.

Reply:
`Mod+Shift` only results in a lowercase letter on Mac AFAIK. [which is what my comment was trying to say](https://github.com/github/hotkey/pull/115#discussion_r1417953735).

Reply:
Well crap, on Windows it results in an uppercase key.

![screenshot showing that windows outputs an uppercase K when pressing control shift k.](https://github.com/github/hotkey/assets/2294248/10700110-214b-4b99-b646-0e35713e27fd)

This is _not_ the conclusion we came to two months ago [when we had this discussion on Slack](https://github.slack.com/archives/C04NALAB3C7/p1697561264365369), and we made some critical decisions based on that conclusion. This is really going to complicate things 🙃

Reply:
![image](https://github.com/github/hotkey/assets/118266/08f1a0a4-c3ad-4eee-bdf2-128a60d23672)

Yes, Mac+Safari _will not_ uppercase the printable character if the `Cmd` key is in the chord. Linux and (I believe) windows _will_ uppercase the printable character if the `Ctrl` key is in the chord. 

![image](https://github.com/github/hotkey/assets/118266/1931208c-ca35-42fb-9d76-05418c64825c)

Linux will also uppercase the printable character if the `Meta` key is in the chord.

Effectively you can consider it that Win+Linux will always uppercase the printable character if the Shift key is in the chord while Mac will only uppercase the printable character for the `Ctrl+Shift` and `Shift` planes (but not `Meta+Shift`.